src/lua/httpc.lua

```diff
@@ -262,13 +264,39 @@ curl_mt = {
             if not method or not url then
                 error('request(method, url[, body, [options]])')
             end
-            local resp = self.curl:request(method, url, body, opts or {})
+            method = method:upper()
+            opts = opts or {}
+            opts.params = opts.params or {}
+            local query_encoder = require("uri")._internal.params
+            local url_with_params = url
+            local ok, res = pcall(query_encoder, opts.params)
+            if (ok == false) then
+                error(res)
+            end
+            local encoded_params = res
+            if encoded_params then
+                if method == "GET" or
+                   method == "HEAD" or
+                   method == "DELETE" then
+                    url_with_params = ("%s?%s"):format(url, encoded_params)
+                elseif body then
+                    error('use either body or http params')
+                else
+                    opts.headers = opts.headers or {}
+                    opts.headers["content-type"] = "application/x-www-form-urlencoded"
```

Comment:
What to do if a user already provided its own content type? Raise an error?

We definitely shouldn't rewrite it.